build: add instructions to rectify code that is not regenerated

otan opened the pull request:
Make it easier for the developer to know how to fix un-generated code by
specifying the required commands.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56004)
<!-- Reviewable:end -->

otan commented:
(failure example: https://github.com/cockroachdb/cockroach/pull/56005)

irfansharif commented on build/teamcity-check.sh:
```diff
@@ -35,22 +37,23 @@ fi
 run build/builder.sh make generate &> artifacts/generate.log || (cat artifacts/generate.log && false)
-run build/builder.sh make buildshort &> artifacts/buildshort.log || (cat artifacts/buildshort.log && false)
 rm artifacts/generate.log
+check_clean "Run \`make generate\` to automatically fix these. For only BUILD.bazel issues, run \`bazel run //:gazelle\`."
```
I'd also be fine with suggesting `make bazel-generate` here.

otan replied:
have to teach people to stop using make somehow :P

otan commented:
made `make bazel-generate` be a separate item, as otherwise it is only triggered on `make vendor_rebuild` and leads to a more confusing message.

otan committed:
build: add instructions to rectify code that is not regenerated

Make it easier for the developer to know how to fix un-generated code by
specifying the required commands.

Release note: None

otan commented:
bors r=irfansharif
